rholang/src/main/k/rholang/free.k

```diff
@@ -137,6 +137,113 @@ rule <thread> ...
          <NodeCounter> (NEWID:Int => NEWID +Int 2) </NodeCounter>
     ... </FreeVars> ... </thread>
 
+
+
+// Logical "and"
+// First for a conjunction of HigherProcesses
+rule <thread> ...
+        <FreeVars> ...
+          <node> [ID:Int][ P1:HigherProc /\ P2:HigherProc ][ PARENT:Exp ][ (.List
+            => ListItem(NEWID) ListItem(NEWID +Int 1)) ]{ FREEVARS:List }[ (false
+            => true) ] </node>
+         (.Bag => <node> [NEWID][ P1 ][ ID ][ .List ]{ .List }[false]  </node>
+                  <node> [NEWID +Int 1][ P2 ][ ID ][ .List ]{ .List }[false]  </node>)
+         <NodeCounter> (NEWID:Int => NEWID +Int 2) </NodeCounter>
+    ... </FreeVars> ... </thread>
+
+
+
+// Logical "or"
+// For a statmenet P1 \/ P2, we do not bind any variables in either P1 or P2. Thus if we hit a
+// logical "or", we check for any free variables at all. Any variables here are free variables
+// in the entire statement.
+
+// For channels
+ // Make the 'ornode'
+rule <thread> ...
+        <FreeVars> ...
+          <node>
+            [ID:Int][ C1:Chan \/ C2:Chan ][ PARENT:Exp ][
```

Comment:
Again, logical connectives in general only apply to process-valued patterns.  Instead of `for (x <- @P \/ @Q) { R }`, we'll write `for (x <- @{P \/ Q}) {R}`.